raft.go

```diff
@@ -552,10 +671,9 @@ func (r *Raft) runFSM() {
 				continue
 			}
 
-			// Get our peers
-			peers, err := r.peerStore.Peers()
-			if err != nil {
-				req.respond(err)
+			if lastIndex < r.configurations.committedIndex {
+				req.respond(fmt.Errorf("cannot take snapshot now, wait until the configuration entry at %v has been applied (have applied %v)",
```

Comment:
This'll be a little weird in the logs but should be exceedingly rare. It will just bubble up and result in a log message and shouldn't break anything.